### PR TITLE
Fix add partition relops null pointer issue.

### PR DESCRIPTION
**Pull request by kainwen**
transformRelOptions may return a null pointer in some
cases, add the check in function `add_partition_rule`.

## Here are some reminders before you submit the pull request
- [ ] Add tests for the change
- [ ] Document changes
- [ ] Communicate in the mailing list if needed
- [ ] Pass `make installcheck`
- [ ] Review a PR in return to support the community

### CODE REVIEW

File: src/test/regress/sql/partition1.sql [hunk 1]
```diff
@@ -1021,6 +1021,27 @@ drop table if exists s1;
...
+);
+
+alter table test_part_relops_tmpl
+set subpartition template
+(
+   subpartition l1 values('l1') with (appendonly=false)
+);
+
+alter table test_part_relops_tmpl alter partition for ('m1') add partition l3 values ('l3')
+with (oids=false);
```
Comment (paul-guo-) [inline src/test/regress/sql/partition1.sql]: Can you explain in test cases about this issue to save time when people look at the test in the future (else need to use git blame...)

**Comment by ashwinstar** [comment]
Do we need to carefully scrutinize other calls to `transformRelOptions()`? I see it called at quite few places where return value is not checked for NULL but instead assigned to values[xxx] like `add_partition_encoding()` and ...

File: src/test/regress/sql/partition1.sql [hunk 2]
```diff
@@ -1021,6 +1021,29 @@ drop table if exists s1;
...
 -- end_ignore
 
+-- the following case is to test when we have a template
+-- we can correct add new subpartition with relation options.
+create table test_part_relops_tmpl (id int,  p1 text, p2 text, count int)
+distributed by (id)
+partition by list (p1)
+subpartition by list (p2)
+(
+  partition m1 values ('m1') with (appendonly=false)
```
[the remark this answers is not in the document]
Comment (kainwen) [reply src/test/regress/sql/partition1.sql]: > Still hard to follow the test. Please lets add comment explaining what significance of `appendonly=false` and `oids=false` in this test. I think the reason for having them is we wish to specify something in reloptions which are defaults and hence `transformRelOptions()` will return NULL, as for default it returns NULL, right?

Add more comments. Thanks for your review.

**Comment by kainwen** [comment]
> Do we need to carefully scrutinize other calls to `transformRelOptions()`? I see it called at quite few places where return value is not checked for NULL but instead assigned to values[xxx] like `add_partition_encoding()` and ...

OK. I will go through other places. Also ping @paul-guo- here.

File: src/test/regress/sql/partition1.sql (hunk 2 above)
[the remark this answers is not in the document]
Comment (kainwen) [reply src/test/regress/sql/partition1.sql]: > Still hard to follow the test. Please lets add comment explaining what significance of `appendonly=false` and `oids=false` in this test. I think the reason for having them is we wish to specify something in reloptions which are defaults and hence `transformRelOptions()` will return NULL, as for default it returns NULL, right?

Hi @ashwinstar 
The C code has been updated a bit.  And I have updated the comments in test also.

Would you please have a look at it again?

Thanks!

[the remark this answers is not in the document]
Comment (ashwinstar) [reply src/test/regress/sql/partition1.sql]: We have still not explained why `appendonly=false` and `oids=false` is used in test. Why the test doesn't use `oids=true` or `appendonly=true`? Comments should always explain WHY, what the test does can be mostly figured out from the flow.

**Comment by kainwen** [comment]
Hi @ashwinstar 
I am going to merge this. Are you satisfied with the comments now?
Thanks.

File: src/test/regress/sql/partition1.sql (hunk 2 above)
[the remark this answers is not in the document]
Comment (ashwinstar) [reply src/test/regress/sql/partition1.sql]: I am still not seeing reason for using `appendonly=false` anywhere in comments?

[the remark this answers is not in the document]
Comment (kainwen) [reply src/test/regress/sql/partition1.sql]: > I am still not seeing reason for using `appendonly=false` anywhere in comments?

Sorry for the misleading.

I have removed `appendonly` option, they are from the escalation but nothing to do with the case itself.
